cmd/end2endtest/account.go

```diff
@@ -1,32 +1,277 @@
 package main
 
 import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/uuid"
+	apipkg "go.vocdoni.io/dvote/api"
+	"go.vocdoni.io/dvote/apiclient"
 	"go.vocdoni.io/dvote/crypto/ethereum"
 	"go.vocdoni.io/dvote/log"
+	"go.vocdoni.io/dvote/vochain"
+	"go.vocdoni.io/dvote/vochain/state"
+	"go.vocdoni.io/proto/build/go/models"
 )
 
-func testTokenTransactions(
-	host,
-	treasurerPrivKey string,
-	keySigner string,
-) {
-	treasurerSigner, err := privKeyToSigner(treasurerPrivKey)
+func testTokenTransactions(c config) {
+	// create alice signer
+	alice := &ethereum.SignKeys{}
+	if err := alice.Generate(); err != nil {
+		log.Fatal(err)
+	}
+
+	// create bob signer
+	bob := &ethereum.SignKeys{}
+	if err := bob.Generate(); err != nil {
+		log.Fatal(err)
+	}
+
+	// Connect to the API host
+	hostURL, err := url.Parse(c.host)
+	if err != nil {
+		log.Fatal(err)
+	}
+	log.Debugf("connecting to %s", hostURL.String())
+
+	token := uuid.New()
+	api, err := apiclient.NewHTTPclient(hostURL, &token)
 	if err != nil {
 		log.Fatal(err)
 	}
-	// create main signer
-	mainSigner := &ethereum.SignKeys{}
-	if err := mainSigner.Generate(); err != nil {
+
+	// check transaction cost
+	if err := testGetTxCost(api); err != nil {
 		log.Fatal(err)
 	}
 
-	// create other signer
-	otherSigner := &ethereum.SignKeys{}
-	if err := otherSigner.Generate(); err != nil {
+	fp, err := getFaucetPackage(c, alice.Address().Hex())
+	if err != nil {
+		log.Fatal(err)
+	}
+	// check create and set account
+	if err := testCreateAndSetAccount(api, fp, alice, bob); err != nil {
 		log.Fatal(err)
 	}
 
-	log.Infof("connecting to main gateway %s", host)
-	log.Fatal("wip, not yet implemented",
-		treasurerSigner)
+	// check send tokens
+	if err := testSendTokens(api, alice, bob); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func testGetTxCost(api *apiclient.HTTPclient) error {
+	// get treasurer nonce
+	treasurer, err := api.Treasurer()
+	if err != nil {
+		return err
+	}
+	log.Infof("treasurer is %s", common.BytesToAddress(treasurer.Address))
```

Comment:
from now on, we are trying to use the key/value log functions when printing information with values, such as `log.Infow("treasurer", map[string]interface{}{"address": common.BytesToAddress(treasurer.Address)}`

Also use `log.Errorw(err, "message")` for printing errors.

Reply:
right, forgot about that. will do :+1: